onnxruntime/contrib_ops/cpu/nchwc_ops.cc

```diff
@@ -0,0 +1,222 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/framework/op_kernel_context_internal.h"
+#include "nchwc_ops.h"
+#include "core/mlas/inc/mlas.h"
+
+namespace onnxruntime {
+namespace contrib {
+
+#define ONNX_CPU_OPERATOR_TYPED_NCHWC_KERNEL(name, ver, type, builder, ...) \
+  ONNX_OPERATOR_TYPED_KERNEL_EX(name, kMSNchwcDomain, ver, type, kCpuExecutionProvider, builder, __VA_ARGS__)
+
+ONNX_CPU_OPERATOR_TYPED_NCHWC_KERNEL(
+    ReorderInput,
+    1,
+    float,
+    KernelDefBuilder()
+        .TypeConstraint("T", DataTypeImpl::GetTensorType<float>()),
+    ReorderInput<float>);
+
+ONNX_CPU_OPERATOR_TYPED_NCHWC_KERNEL(
+    ReorderOutput,
+    1,
+    float,
+    KernelDefBuilder()
+        .TypeConstraint("T", DataTypeImpl::GetTensorType<float>()),
+    ReorderOutput<float>);
+
+ONNX_CPU_OPERATOR_TYPED_NCHWC_KERNEL(
+    Conv,
```

Comment:
Instead of duplicating names why not add unique names for these ops? Like NCHWcConv or NCHWcMaxPool?

Reply:
This was an implementation trick (hack?) to keep using the PoolBase class without change. That code does checks for OpName() against MaxPool, GlobalMaxPool, etc., to control behavior. To avoid changing that, we agreed to put the NCHWc ops in kMSNchwcDomain and allow reuse of the original op names.